src/evict/evict_lru.c

```diff
@@ -1579,13 +1579,12 @@ __evict_get_ref(
 	if (is_server &&
 	    (cache->evict_empty_score > WT_EVICT_SCORE_CUTOFF ||
 	    __evict_queue_empty(cache->evict_fill_queue, false))) {
-		do {
+		while ((ret = __wt_spin_trylock(
+		    session, &cache->evict_queue_lock)) == EBUSY)
 			if ((!urgent_ok ||
 			    __evict_queue_empty(urgent_queue, false)) &&
 			    !__evict_queue_full(cache->evict_fill_queue))
 				return (WT_NOTFOUND);
```

Comment:
I haven't measured the perf impact at all: the case where the queues are completely empty should be caught right at the top of `__evict_get_ref`...